### PR TITLE
Use setDocumentTitle method of JViewLegacy to set document title

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue # .

### Summary of Changes
Since Joomla 3.6 , JViewLegacy has implemented the method setDocumentTitle https://github.com/joomla/joomla-cms/blob/staging/libraries/legacy/view/legacy.php#L854-L873 which is used to set document title base on site configuration. However, in many views in Joomla code base, we still use manual code, repeating code to set page title. This PR solves it

### Testing Instructions
A real test would be check all views changes by this PR and make sure the document title is not changed before and after patch. However, honestly, I don't think we can expect that kind of test here because it would take much time

So I hope developers can help reviewing the code of the PR and confirm that it is expected changes. If you review code, please help reviewing every file carefully.

### Documentation Changes Required
None

### CODE REVIEW

File: components/com_contact/views/contact/view.html.php [hunk 1]
```diff
@@ -334,14 +334,12 @@ public function display($tpl = null)
 	 */
 	protected function _prepareDocument()
 	{
-		$app     = JFactory::getApplication();
-		$menus   = $app->getMenu();
+		$app     = JFactory::getApplication();		
 		$pathway = $app->getPathway();
-		$title   = null;
-
+
```
Comment (zero-24) [inline components/com_contact/views/contact/view.html.php]: please remove the tabs here and just lets this be a clean line.

**Review by zero-24 — CHANGES_REQUESTED** [review]
Please remove the tabs and replace it with clean lines.

File: components/com_contact/views/contact/view.html.php (hunk 1 above)
Comment (joomdonation) [inline components/com_contact/views/contact/view.html.php]: @zero-24: Could I ask for your help with code style issues here?

Comment (zero-24) [inline components/com_contact/views/contact/view.html.php]: what kind of help do you need? 😄

Comment (joomdonation) [inline components/com_contact/views/contact/view.html.php]: The issues with code styles if it is possible :)

Comment (joomdonation) [inline components/com_contact/views/contact/view.html.php]: Any of course if you can help reviewing the changes in the code to make sure there are no mistakes, too :)

Comment (zero-24) [inline components/com_contact/views/contact/view.html.php]: https://github.com/joomdonation/joomla-cms/pull/3

Please try this and than we can look into it again

Comment (joomdonation) [inline components/com_contact/views/contact/view.html.php]: Many thanks for helping.

**Review by zero-24 — APPROVED** [review]

**Comment by zero-24** [comment]
looks good now from a cs persprctive 👍

**Comment by joomdonation** [comment]
Again, many thanks for your help @zero-24. I realize that there are still changes needed to some other files (to remove un-used variables + minor optimization). In case there are CS changes needed, I would love to receive your help again :)

**Comment by zero-24** [comment]
no problem just ping me 😄

**Comment by joomdonation** [comment]
@wilsonge @rdeutz  This PR makes changes to many files, should I spit it into smaller PRs (each PR for one component) for easier code review and we can get human testing, too?

**Comment by rdeutz** [comment]
if you had 200 than yes but 20 changed files is ok

**Comment by joomdonation** [comment]
OK. Thanks. I will leave it as how it is, then

**Review by n9iels — APPROVED** [review]

**Comment by n9iels** [comment]
I have tested this item :white_check_mark: successfully on 823693e9216357c8958560857f6d1a0014e255ff<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14250">issues.joomla.org/tracker/joomla-cms/14250</a>.</sub>

**Comment by n9iels** [comment]
@joomdonation What is the reason to close this PR?

**Comment by joomdonation** [comment]
@n9iels Honestly, a PR which changes multiple files like this would be hard to get testers and proper testings

Also, I was thinking about introduce a new method which set both page title and page meta data (which basically including this block of code as well https://github.com/joomdonation/joomla-cms/blob/0031973e956bc83700140cb905e4a963c195cd45/components/com_content/views/form/view.html.php#L163-L176 )

So I think it is better to close this PR and do proper coding in Joomla 4. Thanks for reviewing the changes though :)

**Pull request closed without merging**